ftw/casauth/www/config.zpt

```diff
@@ -14,7 +14,7 @@
           <div class="form-help">e.g. https://cas.server.net</div>
           </td>
           <td><input type="text" name="cas_server_url"
-                     tal:attributes="value context/cas_server_url"/></td>
+                     tal:attributes="value context/stored_cas_server_url"/></td>
```

Comment:
So the config screen always shows (and allows to change) the stored cas_server_url, even if it won't have any effect because it's overridden by the environment variable which has precedence (same for the internal URL).

This feels like it has the potential to lead to quite frustrating debugging situations.

Could you maybe display the _effective_ URL as well on this form (as a read-only "field")? Even without details about where it's coming from etc.. this would at least allow to notice when it's different from the stored URL, and prompt people to look more closely at how the implementation works in situations where things are not working as expected.